src/test/java/pingis/controllers/TaskControllerTest.java

```diff
@@ -53,84 +58,135 @@
     private ChallengeService challengeServiceMock;
 
     @MockBean
+    private TaskImplementationService taskImplementationServiceMock;
+            
+    @MockBean
     private TaskService taskServiceMock;
 
-    // Never accessed but necessary for testing
+
     @MockBean
     private EditorService editorServiceMock;
 
     @Captor
     private ArgumentCaptor<Map<String, byte[]>> packagingArgCaptor;
+    
+    private Challenge challenge;
+    private Task testTask;
+    private Task implementationTask;
+    private ChallengeImplementation testChallengeImplementation;
+    private TaskImplementation testTaskImplementation;
+    private TaskImplementation implTaskImplementation;
+    private User testUser;
 
     @Before
     public void setUp() {
+        testUser = new User(1, "TESTUSER", 1);
+        challenge = new Challenge("Calculator", testUser,
+                "Simple calculator");
+        testTask = new Task(1,
+                ImplementationType.TEST, testUser, "CalculatorAddition",
+                "Implement addition", "return 1+1;", 1, 1);
+        implementationTask = new Task(2,
+                ImplementationType.IMPLEMENTATION, testUser, "implement addition",
+                "implement addition", "public test", 1, 1);
+        testChallengeImplementation
+                = new ChallengeImplementation(challenge, testUser, testUser);
+        testTaskImplementation
+                = new TaskImplementation(testUser, "", testTask);
+        testTaskImplementation.setChallengeImplementation(testChallengeImplementation);
+        implTaskImplementation = new TaskImplementation(testUser, "",
+                implementationTask);
+        implTaskImplementation.setChallengeImplementation(testChallengeImplementation);
+        challenge.addTask(implementationTask);
+        implementationTask.setChallenge(challenge);
         MockitoAnnotations.initMocks(this);
 
         this.mvc = MockMvcBuilders
                 .webAppContextSetup(context)
                 .build();
     }
 
+
+
     @Test
-    public void submitTask() throws Exception {
+    public void givenTaskWhenGetTestTask() throws Exception {
+        when(taskImplementationServiceMock.findOne(testTaskImplementation.getId()))
+                .thenReturn(testTaskImplementation);
+        Map<String, EditorTabData> tabData = this.generateTestTabData(implTaskImplementation, testTaskImplementation);
+        when(editorServiceMock.generateEditorContents(testTaskImplementation)).thenReturn(tabData);
+        String uri = "/task/"+testTaskImplementation.getId();
+        performSimpleGetRequestAndFindContent(uri, "task", testTask.getCodeStub());
+        verify(taskImplementationServiceMock, times(1)).findOne(testTaskImplementation.getId());
+        verify(editorServiceMock, times(1))
```

Comment:
Calling `verify()` without a second parameter is equal to `verify(..., times(1))`, which can make lines shorter.